lib/account-data.js

```diff
@@ -1,95 +1,123 @@
-const fs = require('fs')
 const path = require('path')
-const rimraf = require('rimraf')
-const os = require("os");
-const {data,appDataPath} = require('./config')
-const userFiles = path.join(os.homedir(),'AppData','Roaming')
-const parsecUser = path.join(userFiles,'Parsec','user.bin')
-const accountData = path.join(appDataPath,'account_data.json')
-const currentAccountFile = path.join(appDataPath,'currentAccount.txt')
-
-const multiUserFolder = path.join(userFiles,'Parsec','UserList')
-
-
+const {global_state} = require("./initialize")
+const {error} = require("./error")
+const {fileFuncs} = require("./file-operations")
+const constants = require("./constants")
 //todo need some rewriting.
 
-
 function checkNickname(nickname){
+
     let exist = 0
-    if(data.has(nickname)){
+    if(global_state.data.has(nickname)){
         exist = 1
     }
     return exist
 }
 
-function removeParsecUserFile(){
-    if (fs.existsSync(parsecUser)){
-        fs.unlinkSync(parsecUser)
+async function removeParsecUserFile(){
+    if (await fileFuncs.checkExists(global_state.config['parsecUser'])){
+        let op = await fileFuncs.deleteFile(global_state.config['parsecUser'])
+        if (!op.success){
+            return error.FILE_DELETE_ERROR
+        }
+        else {
+            return 0
+        }
+        //fs.unlinkSync(global_state.parsecUser)
     }
+    return 0
 
 
 }
-function backupParsec(){
+async function removeAccount(nickname){
+    let userPathWithoutBin = path.join(constants.multiUserPath,nickname)
+    let userPath = path.join(constants.multiUserPath,nickname,'user.bin')
+    if(await fileFuncs.checkExists(userPath)){
+        let op = await fileFuncs.deleteFolder(userPath)
+        if (!op.success){
+            return error.FOLDER_DELETE_ERROR
+        }
+        op = await fileFuncs.deleteFolder(userPathWithoutBin)
+        if(!op.success){
+            return error.FOLDER_DELETE_ERROR
+        }
 
-}
+    }
+    if(nickname === global_state.config.currentUser){
+        let op =  await removeParsecUserFile()
+        if(op){
+            return op
+        }
```

Comment:
`const error` will be more readable than `op`

Reply:
Can't use error as it's already used for something else, so used error1, error2 etc, please check